test/api/core_test.dart

```diff
@@ -8,25 +8,31 @@ import 'package:zulip/api/core.dart';
 import 'package:zulip/api/exception.dart';
 import 'package:zulip/model/localizations.dart';
 
+import '../model/binding.dart';
 import '../stdlib_checks.dart';
 import 'exception_checks.dart';
 import 'fake_api.dart';
 import '../example_data.dart' as eg;
 
 void main() {
+  TestZulipBinding.ensureInitialized();
+  tearDown(testBinding.reset);
+
   test('ApiConnection.get', () async {
     Future<void> checkRequest(Map<String, dynamic>? params, String expectedRelativeUrl) {
-      return FakeApiConnection.with_(account: eg.selfAccount, (connection) async {
-        connection.prepare(json: {});
-        await connection.get(kExampleRouteName, (json) => json, 'example/route', params);
-        check(connection.lastRequest!).isA<http.Request>()
-          ..method.equals('GET')
-          ..url.asString.equals('${eg.realmUrl.origin}$expectedRelativeUrl')
-          ..headers.deepEquals({
-            ...authHeader(email: eg.selfAccount.email, apiKey: eg.selfAccount.apiKey),
-            ...userAgentHeader(),
-          })
-          ..body.equals('');
+      return FakeApiConnection.with_(account: eg.selfAccount,
+        userAgentHeader: testBinding.userAgentHeader(),
+        (connection) async {
+          connection.prepare(json: {});
+          await connection.get(kExampleRouteName, (json) => json, 'example/route', params);
+          check(connection.lastRequest!).isA<http.Request>()
+            ..method.equals('GET')
+            ..url.asString.equals('${eg.realmUrl.origin}$expectedRelativeUrl')
+            ..headers.deepEquals({
+              ...authHeader(email: eg.selfAccount.email, apiKey: eg.selfAccount.apiKey),
+              ...testBinding.userAgentHeader(),
+            })
+            ..body.equals('');
```

Comment:
I think this is a one-line change plus a reformatting with different whitespace; is that right? The whitespace change is kind of distracting to me as I review looking for meaningful changes. (Here and in several places below.)

Reply:
It looks like the whitespace changes in question are all an indentation change. It's natural for those to happen together with substantive changes sometimes — for example they'll happen whenever adding or removing a wrapper widget around a child subtree.

A handy tool for reading such diffs is `git log -b`, which ignores indentation changes. With that, this diff hunk looks like:
```diff
-      return FakeApiConnection.with_(account: eg.selfAccount, (connection) async {
+      return FakeApiConnection.with_(account: eg.selfAccount,
+        userAgentHeader: testBinding.userAgentHeader(),
+        (connection) async {
           connection.prepare(json: {});
           await connection.get(kExampleRouteName, (json) => json, 'example/route', params);
           check(connection.lastRequest!).isA<http.Request>()
             ..method.equals('GET')
             ..url.asString.equals('${eg.realmUrl.origin}$expectedRelativeUrl')
             ..headers.deepEquals({
               ...authHeader(email: eg.selfAccount.email, apiKey: eg.selfAccount.apiKey),
-            ...userAgentHeader(),
+              ...testBinding.userAgentHeader(),
             })
             ..body.equals('');
```

Reply:
(With `git config diff.colorMovedWS`, you can also get your default `git log -p` output to treat indentation changes the same way as moved code and give them a different background color. That's what I have [in my gitconfig](https://gist.github.com/gnprice/5f71b63464a76bd7e0480f207b1bbff3#file-gitconfig-L91-L119). I still use `-b` to get a clearer view in extreme cases.)

Reply:
Cool. Yeah; I used `git log -b` to help verify the change, but I think I could have looked closer at the regular `git log -p` output—it turns out I already have that git config option set the way you do, and here's what it looks like:

<img width="710" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/3d478bdf-e61e-4863-9d3d-5cfc9cdb023b">

Reply:
Yeah, exactly. And yeah, those colors are adjustable - they're the diff.oldMoved and friends found in my gitconfig. Definitely worth trying some quick changes to them to see what works best for you.

From the screenshot, it looks like your terminal's default background color is already a faint gray rather than black, which ends up being pretty close to the background colors set there. I have my terminal background set to pure black, for maximum contrast, and then the colors in my config are chosen to be distinct from that.

Reply:
This is better—I've changed the Android Studio color scheme to "high contrast":

<img width="728" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/cd85622e-39dd-4520-b067-7648fbcf7414">

Reply:
Yeah, here's what that looks like for me:
![image](https://github.com/zulip/zulip-flutter/assets/28173/de35ce05-66cd-4de1-8fa7-6310cbab3526)


The striping in your screenshot looks like it could get a bit annoying. You might be able to adjust the line spacing to fix that — generally I'd expect a terminal to use 1.0x line spacing anyway.

Reply:
> The striping in your screenshot looks like it could get a bit annoying. You might be able to adjust the line spacing to fix that — generally I'd expect a terminal to use 1.0x line spacing anyway.

Yep, got it! I adjusted a font setting in the Android Studio settings. It was a little confusing because things outside the terminal were responding to the setting, but the terminal wasn't—but then it did after I restarted Android Studio.